avocado/utils/file_utils.py

```diff
@@ -0,0 +1,89 @@
+"""
+
+SUMMARY
+------------------------------------------------------
+Utilities for file tests.
+
+Copyright: Intra2net AG
+
+
+INTERFACE
+------------------------------------------------------
+
+"""
+
+from glob import iglob
+import os
+from pwd import getpwnam, getpwuid
+from grp import getgrnam, getgrgid
+from stat import S_IMODE
+import hashlib
+import logging
+
+
+def check_owner(owner, group, file_name_pattern, check_recursive=False):
+    """
+    Verifies that given file belongs to given owner and group.
+
+    :param str owner: user that owns of the file
+    :param str group: group of the owner of the file
+    :param str file_name_pattern: can be a glob
+    :param check_recursive: if file_name_pattern matches a directory,
+                            recurse into that subdir or not
+    :type check_recursive: bool
+    :raises: :py:class:`RuntimeError` if file has wrong owner or group
+    """
+    for file_name in iglob(file_name_pattern):
+        actual_id = os.stat(file_name).st_uid
+        if actual_id != getpwnam(owner).pw_uid:
+            raise RuntimeError(
+                'file {0} has wrong owner {1} (should be {2})'
+                .format(file_name, getpwuid(actual_id).pw_name, owner))
+        actual_id = os.stat(file_name).st_gid
+        if actual_id != getgrnam(group).gr_gid:
+            raise RuntimeError(
+                'file {0} has wrong group {1} (should be {2})'
+                .format(file_name, getgrgid(actual_id).gr_name, group))
+        logging.debug('checked owner {0}:{1} of file {2}'
+                      .format(owner, group, file_name))
+
+        if check_recursive and os.path.isdir(file_name):
+            new_pattern = os.path.join(file_name,
+                                       os.path.basename(file_name_pattern))
+            check_owner(owner, group, new_pattern, True)
+
+
+def check_permissions(perms, file_name_pattern):
+    """
+    Verify that a given file has a given numeric permission.
+
+    :param int perms: best given in octal form, e.g. 0o755
+    :param str file_name_pattern: can be a glob
+    :raises: :py:class:`RuntimeError` if file has wrong permissions
+    """
+    for file_name in iglob(file_name_pattern):
+        actual_perms = S_IMODE(os.stat(file_name).st_mode)
+        if perms != actual_perms:
+            raise RuntimeError(
+                'file {0} has permissions {1} (should be {2})!'
+                .format(file_name, oct(actual_perms), oct(perms)))
+        logging.debug('checked permissions {0} of file {1}'
+                      .format(oct(perms), file_name))
+
+
+def hash_file(filepath, algo=None, bs=4096):
+    """
+    Verify file contents through a MD5 or other hash.
+
+    :param str filepath: full file path to verify
+    :param algo: hash algorithm to use (default MD5)
+    :type algo: :py:class:`_hashlib.HASH`
+    :param int bs: bytes to read and use for the hash
+    """
+    algo = hashlib.md5() if algo is None else aglo
```

Comment:
There's a typo here: `aglo`

Reply:
Fixed, will move to more of your comments later today.